core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -167,7 +166,7 @@
     private final MeanMetric refreshMetric = new MeanMetric();
     private final MeanMetric flushMetric = new MeanMetric();
 
-    private final ShardEngineFailListener failedEngineListener = new ShardEngineFailListener();
+    private final ShardEngineFailListener engineEventListener = new ShardEngineFailListener();
```

Comment:
nit: since we're moving away from the word fail in the name, can we rename `ShardEngineFailListener` to `ShardEngineEventListener`